lib/diskoverflow/slice.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (C) 2018 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package diskoverflow
+
+import (
+	"fmt"
+)
+
+type Slice interface {
+	Append(v Value)
+	NewIterator(reverse bool) Iterator
```

Comment:
I'd prefer to separate this out as NewIterator / NewReverseIterator as it's not particularly clear at the call site that NewIterator(true) means something reverse.